### DISCUSSION TITLE
Would this work for checking if delivery is possible?

### DISCUSSION THREAD

**Discussion by jkatsnelson** [Ephigenia/ikea-availability-checker#34] [Q&A] [answered]
I have a use case where I want to see if a product will become available for delivery. I'd like to check it every day. Do you think the existing functionality would cover this somehow, or should I add something? Any sense for how it would be done? Happy to add it if so.

**Comment by Ephigenia** [comment, marked as answer]
The `stock` sub-command enables searching for multiple stores and products and could use the exit-code as a hint for all (or just a single) products being available or not. I’m not sure if that would be good practice for a CLI tool to misuse the exit code that way.

That’s why I’d suggest using an additional CLI tool for checking the output of the `stock` command as it gives you much more flexibility on how you want to check what.

So for simply checking if a single product is availble you could use the json output and [jq](https://stedolan.github.io/jq/) (one of the standard tools for json processing in the cli) to check the `availability.stock` value like so:

```
npm run start -s -- stock --store=155 --reporter json S69022537 | jq '.[] | .availability | .stock > 0'
```

I had to modify a little part in the response processing that `stock` actually becomes a number and jq can run the comparison. Please use the [latest version of ikea-availability-checker](https://github.com/Ephigenia/ikea-availability-checker/releases/tag/v0.6.8).

The other part of your question is for checking in an interval. I’d suggest using common CLI tools for that too like [watch](https://en.wikipedia.org/wiki/Watch_(command)), [cron](https://en.wikipedia.org/wiki/Cron). In case you’re on OSX there are also ways to [trigger a Desktop Notification](https://code-maven.com/display-notification-from-the-mac-command-line. There are even tools for that on each operation system I guess. 

I hope this may help you.

Maybe parts of this should be added to the README.md - what do you think?

**Reply by jkatsnelson** [reply]
Thanks for the response :-D

Actually, I think I didn’t ask my question very well. I should simplify to: How would you go about checking if something is _available to ship_ vs _available in a store or many stores_? Do you happen to know if you can basically extrapolate that if it’s in stock in a country in some store, then it is in stock to ship?

And then, I was saying if there was any functionality around whether a product is _available to ship_ worth putting in the CLI, I’d be happy to help with that.

Maybe that makes more sense? :)

**Reply by Ephigenia** [reply]
Unfortunately, the IOWS API only provides information about how many items of a product are available in a specific store. There also was a similar request like yours in the issues: https://github.com/Ephigenia/ikea-availability-checker/issues/8

It might work with the newer API here: https://api.ingka.ikea.com/cia/availabilities/ru/de?itemNos=20295930&expand=StoresList,Restocks which returns availabilities for a single product in all of a countries stores. There needs to be additional work to reverse engineer which parts of the response result in which status of which store.

I guess the important fields of the response body are:

- `availabilities[].availableForCashCarry`
- `availabilities[].availableForClickCollect`
- `availabilities[].buyingOption.cashCarry.availability`
- `availabilities[].buyingOption.quantity`

```json
{
  "availabilities": [
  {
      "availableForCashCarry": true,
      "availableForClickCollect": true,
      "buyingOption": {
        "cashCarry": {
          "availability": {
            "probability": {
              "thisDay": {
                "colour": {
                  "rgbDec": "0,255,0",
                  "rgbHex": "00FF00"
                },
                "messageType": "HIGH_IN_STOCK"
              },
              "updateDateTime": "2021-02-22T05:43:37.085Z"
            },
            "quantity": 14,
            "restocks": [
              {
                "earliestDate": "2021-02-23",
                "latestDate": "2021-02-23",
                "quantity": 31,
                "reliability": "HIGH",
                "type": "INTERNAL",
                "updateDateTime": "2021-02-22T05:43:37.085Z"
              }
            ],
            "updateDateTime": "2021-02-22T05:43:37.085Z"
          },
          "range": {
            "inRange": true
          },
          "unitOfMeasure": "PIECE"
        },
        "clickCollect": {
          "range": {
            "inRange": true
          }
        },
        "homeDelivery": {
          "range": {
            "inRange": true
          }
        }
      },
      "classUnitKey": {
        "classUnitCode": "063",
        "classUnitType": "STO"
      },
      "itemKey": {
        "itemNo": "20295930",
        "itemType": "ART"
      }
    }
]
```

**Reply by Ephigenia** [reply]
I’ve created another discussion to reverse-engineer the api: https://github.com/Ephigenia/ikea-availability-checker/discussions/36
